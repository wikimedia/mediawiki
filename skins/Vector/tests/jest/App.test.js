const VueTestUtils = require( '@vue/test-utils' );
const App = require( '../../resources/skins.vector.search/App.vue' );

const defaultProps = {
	id: 'searchform',
	searchAccessKey: 'f',
	searchTitle: 'search',
	showThumbnail: true,
	showDescription: true,
	highlightQuery: true,
	searchPlaceholder: 'Search MediaWiki',
	searchQuery: ''
};

const mount = ( /** @type {Object} */ customProps ) => {
	return VueTestUtils.shallowMount( App, {
		props: Object.assign( {}, defaultProps, customProps ),
		global: {
			mocks: {
				$i18n: ( /** @type {string} */ str ) => ( {
					text: () => str
				} )
			},
			directives: {
				'i18n-html': ( el, binding ) => {
					el.innerHTML = `${binding.arg} (${binding.value})`;
				}
			}
		}
	} );
};

describe( 'App', () => {
	it( 'renders a typeahead search component', () => {
		const wrapper = mount();
		expect(
			wrapper.element
		).toMatchSnapshot();
	} );
} );
