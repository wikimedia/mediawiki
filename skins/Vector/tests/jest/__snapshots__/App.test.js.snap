// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a typeahead search component 1`] = `
<cdx-typeahead-search-stub
  accesskey="f"
  aria-label="Search MediaWiki"
  autoexpandwidth="false"
  buttonlabel="searchbutton"
  class="vector-typeahead-search"
  debounceinterval="120"
  formaction=""
  highlightquery="true"
  id="searchform"
  initialinputvalue=""
  placeholder="Search MediaWiki"
  searchfooterurl=""
  searchresults=""
  searchresultslabel="searchresults"
  showthumbnail="true"
  title="search"
/>
`;
